pkg/workloads/cortex/client/cortex/client.py

```diff
@@ -33,7 +38,134 @@ def __init__(self, env: str):
+
+            if len(conda_dependencies) > 0:
+                with open(project_dir / "conda-packages.txt", "w") as conda_file:
+                    conda_file.write("\n".join(conda_dependencies))
+
+            if config.get("predictor") is None:
+                raise ValueError("`predictor` key must be defined in the `config` dictionary")
+
+            with open(project_dir / "predictor.pickle", "wb") as pickle_file:
+                dill.dump(cls, pickle_file)
+                config["predictor"]["path"] = "predictor.pickle"
+
+            return self.deploy_project(config, project_dir, wait=wait, force=force)
+
+    # CORTEX_VERSION_MINOR x3
+    def deploy_project(
+        self,
+        config: Optional[dict] = None,
+        project_dir: Optional[str] = None,
+        config_file: Optional[str] = None,
+        force: bool = False,
+        wait: bool = False,
+    ) -> list:
+        """
+        Deploy an API from a python project.
+
+        Args:
+            config: A dictionary defining a single Cortex API. Specify either this paramater or the `config_file` paramater but not both.
+            Schema can be found here:
+                → Realtime API: https://docs.cortex.dev/v/master/deployments/realtime-api/api-configuration
+                → Batch API: https://docs.cortex.dev/v/master/deployments/batch-api/api-configuration
+                → Traffic Splitter: https://docs.cortex.dev/v/master/deployments/realtime-api/traffic-splitter
+            project_dir: Directory to a Python project containing your predictor implementation. Required if `config` is specified.
+            config_file: Local path to a yaml file defining Cortex APIs. Specify either this paramater or the `config` paramater but not both.
+            force: Override any in-progress api updates.
+            wait: Streams logs until the APIs are ready.
+
+        Returns:
+            Deployment status, API specification, and endpoint for each API.
+        """
+
+        if config is not None and config_file is not None:
+            raise ValueError(
+                "`config` and `config_file` parameters cannot be specified at the same time, please choose one"
+            )
+
+        if config_file is not None:
+            return self._deploy(config_file, force, wait)
+
+        if config is not None:
+            if project_dir is None:
+                raise ValueError(
+                    "`project_dir` parameter must be provided when using the `config` paramater"
+                )
+
+            if type(config) is not dict:
+                raise ValueError(
+                    "`config` parameter must be a dictionary defining a single Cortex API"
+                )
+
+            if not os.path.exists(project_dir):
+                raise ValueError(f"{project_dir} is not a valid directory")
+
+            cortex_yaml_path = os.path.join(project_dir, f".cortex-{uuid.uuid4()}.yaml")
```

Comment:
If this `deploy_project` is run twice in a row without modifying the API spec and by only specifying the `config` parameter, will this lead to the creation of 2 `yaml` configs? If there are indeed 2 configs created, should we only generate one instead?

Maybe generating a hash based on the input `config` parameter is what we need. What is your opinion?